### PR TITLE
Update addEvent rule evaluation to emit multiple chainAuthArgs for supporting multiple permissions

### PR DESCRIPTION
**Pull request by clemire**
This is a prerequisite for react_reply, which requires allowing channel message writes if the user has WRITE or REACT_REPLY.

### CODE REVIEW

File: core/node/rpc/add_event.go [hunk 1]
```diff
@@ -77,11 +78,20 @@ func (s *Service) addParsedEvent(
 		return err
 	}
 
-	if chainAuthArgs != nil {
-		isEntitled, err := s.chainAuth.IsEntitled(ctx, s.config, chainAuthArgs)
-		if err != nil {
-			return err
+	if len(chainAuthArgsList) > 0 {
+		var isEntitled bool = false
```
Comment (bas-vk) [inline core/node/rpc/add_event.go:82]: just do `isEntitled := false`

File: core/node/rpc/add_event.go [hunk 2]
```diff
@@ -93,11 +103,19 @@ func (s *Service) addParsedEvent(
...
 				}
 			}
+			var chainAuthArgsListStringBuilder strings.Builder
+			chainAuthArgsListStringBuilder.WriteString("[\n")
+			for _, chainAuthArgs := range chainAuthArgsList {
+				chainAuthArgsListStringBuilder.WriteString("  ")
+				chainAuthArgsListStringBuilder.WriteString(chainAuthArgs.String())
+				chainAuthArgsListStringBuilder.WriteString(",\n")
+			}
+			chainAuthArgsListStringBuilder.WriteString("]")
```
Comment (bas-vk) [inline core/node/rpc/add_event.go:113]: Maybe there is a good reason for but this looks a bit fishy. Why not leave the formatting to the logger?

File: core/node/rules/rule_builder_ae.go [hunk 3]
```diff
@@ -127,12 +136,12 @@ func (re *ruleBuilderAEImpl) run() (bool, *auth.ChainAuthArgs, *AddEventSideEffe
 	if err != nil {
 		return false, nil, nil, err
 	}
-	if len(re.checks) == 0 && chainAuthArgs == nil && requiredParentEvent == nil {
+	if len(re.checks) == 0 && chainAuthArgsList == nil && requiredParentEvent == nil {
```
Comment (bas-vk) [inline core/node/rules/rule_builder_ae.go:139]: You probably want to check on `len(chainAuthArgsList) == 0`? If not this check also passed with an empty slice.

**Review by bas-vk — COMMENTED** [review]
I'm not familiar enough with the entitlement checks to review the functional changes so I only looked at the code.

File: core/node/rpc/add_event.go (hunk 2 above)
Comment (clemire) [reply core/node/rpc/add_event.go:113]: You know, I never did figure out how toString works with go - I think you're right, there's already a string method on ChainAuthArgs so this is probably unnecessary.

File: core/node/rules/rule_builder_ae.go (hunk 3 above)
Comment (clemire) [reply core/node/rules/rule_builder_ae.go]: Yes, you're right - thank you for catching this!

**Review by brianathere — APPROVED** [review]

File: core/node/rpc/add_event.go [hunk 4]
```diff
@@ -77,11 +77,20 @@ func (s *Service) addParsedEvent(
 		return err
 	}
 
-	if chainAuthArgs != nil {
-		isEntitled, err := s.chainAuth.IsEntitled(ctx, s.config, chainAuthArgs)
-		if err != nil {
-			return err
+	if len(chainAuthArgsList) > 0 {
+		isEntitled := false
+		var err error
```
Comment (texuf) [inline core/node/rpc/add_event.go:82]: This looks good. Random nit - hard to see on mobile, is there a reason you predicated this var err here? Looks like you return it immediately if it gets set

Comment (texuf) [reply core/node/rpc/add_event.go:82]: *pre-declared

**Pull request merged**